common/analysis/lint_rule_status.h

```diff
@@ -221,6 +240,15 @@ class LintStatusFormatter {
                        absl::string_view url,
                        absl::string_view rule_name) const;
 
+  // Substitute the markers `@` with tokens location
+  // this allows us to create custom reason msg
+  // with different token location that are related to found
+  // vulnerable token
+  std::string ReplaceWithHelperTokens(const std::vector<TokenInfo>& tokens,
```

Comment:
I'd probably call this `FormatWithRelatedTokens()` or similar to emphasize that this is a string formatting functionality.

Also, this needs documentation about the other parameters (And `reason` should probably be `message` to emphasize the formatting part).

The fact that backslash escapes the at-symbol `\@` is missing in the documentation.

Finally, since this is not touching anything in the struct, I'd make it a static method, which makes it easier to identify as a helper function.

Reply:
I have applied the naming changes, but there was a problem with making the method static. Well, it requires the access to `line_column_map` which is not static, causing a problem.